### ISSUE TITLE
x.com: ads

### ISSUE THREAD

**Issue by BRR34L0RD** [uBlockOrigin/uAssets#23732]
### Prerequisites

- [X] This is **NOT** a [_YouTube_](https://www.github.com/uBlockOrigin/uAssets/issues/20586), [_Facebook_](https://www.github.com/uBlockOrigin/uAssets/issues/3367), [_Twitch_](https://www.github.com/uBlockOrigin/uAssets/issues/5184) or [_a shortener/hosting site_](https://github.com/uBlockOrigin/uAssets/discussions/17361) report. _These sites MUST be reported by clicking their **respective links**_.
- [X] I read and understand the [policy about what is a valid filter issue](https://github.com/uBlockOrigin/uAssets/blob/master/README.md#uassets).
- [X] I verified that this issue is not a duplicate. (Use this [button](https://user-images.githubusercontent.com/585534/146582579-c32707a0-36a1-4cc5-ad50-83172c9f67a8.png) to find out.)
- [X] I did not remove any of the [default filter lists](https://github.com/uBlockOrigin/uAssets/assets/66517106/6e8d024a-5869-47cf-b142-b2357eb6c7d0), or I have verified that the issue was not caused by removing any of the default lists.
- [X] I did not enable additional filter lists, or I have verified that the issue still occurs without enabling additional filter lists.
- [X] I do not have custom filters/rules, or I have verified that the issue still occurs without custom filters/rules.
- [X] I am not using uBlock Origin along with other content blockers.
- [X] I have verified that the web browser's built-in content blocker/tracking protection, [network wide/DNS blocking](https://en.wikipedia.org/wiki/DNS_blocking), or my VPN is not causing the issue.
- [X] I have verified that other extensions are not causing the issue.
- [X] If this is about a breakage or detection, I have verified that it is caused by uBlock Origin and isn't a site issue.
- [X] I did not answer truthfully to **ALL** the above checkboxes.

### URL address of the web page

`https://x.com/i/verified-get-verified`

### Category

ads

### Description

An annoying nag screen appears whenever I block a sponsored tweet.

### Other extensions used

Nothing.

### Screenshot(s)

<details>
<summary>Screenshot(s)</summary>

![image](https://github.com/uBlockOrigin/uAssets/assets/129367686/b8496e20-32dc-4d96-a592-2b6292e106b6)

</details>

### Configuration

<details>

```yaml
uBlock Origin: 1.57.2
Firefox: 126
filterset (summary):
 network: 136328
 cosmetic: 50694
 scriptlet: 19848
 html: 1787
listset (total-discarded, last-updated):
 default:
  user-filters: 53-14, never
  ublock-filters: 38100-110, now
  ublock-badware: 8579-0, now
  ublock-privacy: 909-2, now
  ublock-unbreak: 2333-2, now
  ublock-quick-fixes: 195-5, now
  easylist: 87463-160, now
  easyprivacy: 50919-67, now
  urlhaus-1: 12505-2, now
  plowe-0: 3729-1160, now
  spa-1: 5634-88, now
filterset (user): [array of 53 redacted]
trustedset:
 added: [array of 6 redacted]
switchRuleset:
 added: [array of 1 redacted]
userSettings: [none]
hiddenSettings: [none]
supportStats:
 allReadyAfter: 718 ms (selfie)
 maxAssetCacheWait: 105 ms
 cacheBackend: indexedDB
popupPanel:
 blocked: 4
 network:
  ads-twitter.com: 1
  google-analytics.com: 1
  twitter.com: 2
```
</details>

**Comment by loadstring1** [comment]
ublock filters need to be updated to x.com

they recently redirected from twitter.com to x.com domain

**Comment by stephenhawk8054** [comment]
> ublock filters need to be updated to x.com
> 
> they recently redirected from twitter.com to x.com domain

It's already updated

https://github.com/uBlockOrigin/uAssets/commit/f93e44fb19bb8391eb3a0d0d97034277af3ed3bf

**Comment by loadstring1** [comment]
> > ublock filters need to be updated to x.com
> > they recently redirected from twitter.com to x.com domain
> 
> It's already updated: [f93e44f](https://github.com/uBlockOrigin/uAssets/commit/f93e44fb19bb8391eb3a0d0d97034277af3ed3bf)

uBlock filters – Cookie Notices needs to be updated.
![Zrzut ekranu 2024-05-17 144647](https://github.com/uBlockOrigin/uAssets/assets/156520308/abe3579c-9210-47f8-85d6-bae0e44fb547)

**Comment by loadstring1** [comment]
nevermind that also got fixed

my ublock origin didn't catch that change which is very weird because i refreshed cache for this filter list recently

**Comment by Yuki2718** [comment]
@gorhill Not related to this issue, but Can you publish a new version of uBOL ASAP for the domain change?

**Comment by gorhill** [comment]
> having no way to quickly fix mega-sites like Twitter, youtube is not good mentally

The fixes to Twitter mostly relate to cosmetic filters, these cannot be updated dynamically (neither scriptlets). MV3 dynamic filters is only for network filters, not very useful when a big share of filter changes are cosmetic- or scriptlet-based filters.

**Comment by BRR34L0RD** [comment]
As I said, I saw a sponsored tweet, I tried to block the user (in reticenses menu), and after this, the nag screen (as shown in screenshot) appears. I will do more tests, if I won't respond, the issue was solved.

**Comment by Yuki2718** [comment]
> The fixes to Twitter mostly relate to cosmetic filters, these cannot be updated dynamically (neither scriptlets). MV3 dynamic filters is only for network filters, not very useful when a big share of filter changes are cosmetic- or scriptlet-based filters.

As DNR is only about network filters, why can cosmetic filters not be updated dynamically?

**Comment by gorhill** [comment]
> why can cosmetic filters not be updated dynamically?

Because they are injected declaratively through [`scripting.registerContentScript()`](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/scripting/registerContentScripts).

Sure, they can be injected dynamically but this would require the following sequence to take place:

- Page loads
- Worker wakes up if not active: **this adds delay along with (possibly severe) CPU and memory spike**
- Find matching cosmetic filters
- Apply matching filters

The big issue is the wake up of the extension's service worker, which adds delay, which will produce a delay before the cosmetic filters are applied. I chose to go the reliable route for cosmetic and scriplet filtering by relying completely on the declarative approach. I demonstrate here the delay and CPU-/memory-spike issue: https://github.com/uBlockOrigin/uBOL-home/wiki/Frequently-asked-questions-(FAQ)#is-ubo-lite-a-bad-faith-attempt-at-converting-ubo-to-mv3 -- I can add the same sort of demonstration for cosmetic filters.

Also, the non-declarative approach requires to put in all the code responsible for compiling/storing filters, a sizeable amount of code.

I chose the reliable and lean approach, hence the limitations. Whoever wants more than what uBO Lite offers will have to find another solution, uBO Lite is not for them.
